### PR TITLE
feat(scanner): make Clair and Trivy as reserved name for scanners

### LINKED ISSUE #11524: Got four scanners after enabling TLS
image

### PR DESCRIPTION
**Pull request by heww**
Closes #11524

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

File: src/pkg/scan/init.go [hunk 1]
```diff
@@ -19,8 +19,6 @@ import (
 	"github.com/goharbor/harbor/src/lib/q"
 	"github.com/goharbor/harbor/src/pkg/scan/dao/scanner"
 	sc "github.com/goharbor/harbor/src/pkg/scan/scanner"
-	"github.com/goharbor/harbor/src/pkg/types"
-	"github.com/google/uuid"
 	"github.com/pkg/errors"
```
Comment (wy65701436) [inline src/pkg/scan/init.go:22]: use lib/errors instead?

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql [hunk 2]
```diff
@@ -200,3 +200,7 @@ ALTER TABLE replication_task ALTER COLUMN dst_resource TYPE varchar(512);
 /*remove count from quota hard and quota_usage used json*/
 UPDATE quota SET hard = hard - 'count';
 UPDATE quota_usage SET used = used - 'count';
+
+/* make Clair and Trivy as reserved name for scanners in-tree */
+UPDATE scanner_registration SET name = concat_ws('-', name, uuid) WHERE name IN ('Clair', 'Trivy') AND immutable = FALSE;
```
Comment (wy65701436) [inline make/migrations/postgresql/0030_2.0.0_schema.up.sql:205]: we do need to document it as the name will be changed if user is using the external clair or trivy.

Comment (heww) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql:205]: This will be in docs, maybe also in release note.

**Comment by renmaosheng** [comment]
what if pre-v2.0 version has Clair and Trivy installed? how is the upgrade behaving?

**Comment by heww** [comment]
> what if pre-v2.0 version has Clair and Trivy installed? how is the upgrade behaving?

The scanner named Clair or Trivy which was installed manually will be renamed to `Clair-xxx-xxx` or `Trivy-xxx-xxx` (xxx-xxx is the uuid of the scanner).

The scanner named Clair which was installed by harbor will keep the same name as before.

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 2 above)
Comment (steven-zou) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql:205]: @heww 

release notes may be more proper.

**Pull request merged**